src/Select.purs

```diff
@@ -353,9 +338,9 @@ component =
           "ArrowUp"   -> preventIt *> eval' (highlight Prev)
           "ArrowDown" -> preventIt *> eval' (highlight Next)
           "Escape"    -> do
-            st <- getState
+            inputElement <- H.getHTMLElementRef $ H.RefLabel "select-input"
```

Comment:
I wonder if we should move the `RefLabel` to the state just to serve as a const. That way we're not manually typing the label string out whenever we need it, since the compiler isn't going to tell us if we mis-typed it

Reply:
I also wonder if there's much of a performance cost to running `getHTMLElementRef` every time we need the input element

Reply:
Yeah, it'd be nice to have this as a string constant, but having it on the state seems like a heavy price for that since we'd have to have access to the state in those setProps functions, which just felt kinda ugly when I had it in here before.

As for `getHTMLElementRef`, I have to imagine there's some cost to querying the DOM as opposed to just storing the element on state, but it will only be happening on focus, blur, or an `ESC` keypress so I doubt it will be a problem. We can always go back to storing it on state and fetching it during initialization if it does become an issue, but then we may have to think about what happens if that element leaves the DOM and returns at some point.

Reply:
Yeah, it'll be something to look into if there's a noticeable lag for any of those actions, otherwise I don't think it's a big deal.

As for the `RefLabel` on the state, I just meant updating the state to something like:
```purescript
type State item =
  { ...
  , refLabel :: RefLabel
  }

--

initialState i = store i.render
      { ...
      , refLabel :: RefLabel "select-input"
      }
```

And then the `Focus` and `Key` evals use that instead of recreating the string. Don't even need it on the state, could just make it a cons inside the component's `where` block